src/background/actions/torrentTracker.ts

```diff
@@ -0,0 +1,70 @@
+import Axios from "axios";
+import bencodec from "bencodec";
+import { startsWithAnyProtocol, MAGNET_PROTOCOL } from "../../common/apis/protocols";
+import type { State } from "../../common/state";
+
+let cachedTrackers: string[] = [];
+let lastPublicTrackerURL = "";
+
+async function updateRemoteTrackers(url: string) {
+  let response;
+
+  try {
+    response = await Axios.get(url, { timeout: 10000 });
+    lastPublicTrackerURL = url;
+  } catch (e) {
+    console.log("Axios Error caught when updating public trackers:", e);
+    cachedTrackers = [];
+  }
+
+  const trackerText: string = response?.data?.toString();
+
+  if (trackerText !== "") {
+    if (trackerText.includes(",")) {
+      cachedTrackers = trackerText.split(",");
+    } else if (trackerText.includes("\n\n")) {
+      cachedTrackers = trackerText.split("\n\n");
+    } else {
+      cachedTrackers = trackerText.split("\n");
+    }
+    console.log("successfully updated public trackers:", cachedTrackers.length);
+  }
+}
+
+export function updateAndGetTorrentTrackers(storedState: State): string[] {
+  console.debug("updateAndGetTorrentTrackers was called", new Error().stack);
+  console.debug("cached trackers:", cachedTrackers.length);
+
+  const flag = storedState.settings.torrentTrackers.enablePublicTrackers;
```

Comment:
I think there's a bug around this flag.

1. User enables public trackers.
2. The extension fetches and holds onto the list.
3. User disables public trackers.
4. User adds a new torrent task.
5. Public trackers are added anyway.

I think something should clear the list if the flag is unset.

Reply:
Will fix later